packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -190,7 +190,8 @@ class AndroidDevice extends Device {
   @override
   bool isAppInstalled(ApplicationPackage app) {
     // This call takes 400ms - 600ms.
-    if (runCheckedSync(adbCommandForDevice(<String>['shell', 'pm', 'path', app.id])).isEmpty)
+    String listOut = runCheckedSync(adbCommandForDevice(<String>['shell', 'pm', 'list', 'packages']));
```

Comment:
Add app.id to the command line.

This will reduce the output of "pm list packages" to only packages that contain app.id in their names